react/src/pages/utils/components/GridFieldsDisplay.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { Grid, GridProps } from "@material-ui/core";
+import { Field } from "../typings";
+import FieldDisplayEditable from "./FieldDisplayEditable";
+
+// Grids are very picky about what sizes are allowed
+type Width = Exclude<GridProps["xs"], boolean | "auto" | undefined>;
+
+/**
+ * Display a collection of editable fields. Number of columns can be specified.
+ *
+ * @param orderPriority How to display the fields (left-right first or top-down first)
+ * @param columnWidth The width of each column (out of 12, must be an integer)
+ */
+export default function GridFieldsDisplay(props: {
+    fields: Field[];
+    editMode: boolean;
+    onEdit: (fieldName: string | undefined, value: any) => void;
+    enums: any;
+    orderPriority?: "left-right" | "top-down";
+    columnWidth?: Width;
+}) {
+    const infoWidth = props.columnWidth || 6;
+    const numColumns = Math.floor(12 / infoWidth);
+
+    let fieldColumns: Field[][] = [...Array(numColumns)].map(() => []);
```

Comment:
You don't need to copy an array that you just constructed and are discarding anyway for the `map`

Reply:
`Array(numColumns)` only sets the `length` property but doesn't put any elements in the array, so `map` doesn't work. `[...Array(numColumns)]` has `numColumns` undefined elements in it, so `map` works (`Array(x).fill(undefined)` would also work). Just JS things...

Reply:
Yeah, and we can't do `Array(x).fill([])` or the same array will be in every slot 🔥